File: packages/gbnf-experiment/data/20260909T005215Z_9a3e20fe/ported_implementation/test/parseSpace.test.ts
import assert from 'node:assert/strict';
import { describe, it } from 'node:test';

import { parseSpace } from '../src/rulesBuilder/parseSpace.ts';

describe('parseSpace', () => {
  it('returns the input string when there is no whitespace or comments', () => {
    const inputStr = 'abcdefghijk';
    const pos = parseSpace(inputStr, 0, true);
    assert.equal(inputStr.slice(pos), inputStr);
  });

  it('skips leading spaces and tabs', () => {
    const inputStr = '   \t   abcdefghijk';
    const pos = parseSpace(inputStr, 0, true);
    assert.equal(inputStr.slice(pos), 'abcdefghijk');
  });

  it('skips leading newline characters when newlineOk is true', () => {
    const inputStr = '\n\n\r\n\r\nabcdefghijk';
    const pos = parseSpace(inputStr, 0, true);
    assert.equal(inputStr.slice(pos), 'abcdefghijk');
  });

  it('does not skip leading newline characters when newlineOk is false', () => {
    const inputStr = '\n\n\r\n\r\nabcdefghijk';
    const pos = parseSpace(inputStr, 0, false);
    assert.equal(inputStr.slice(pos), inputStr);
  });

  it('skips comments and leading spaces', () => {
    const inputStr = '  # This is a comment\n\t   abcdefghijk';
    const pos = parseSpace(inputStr, 0, true);
    assert.equal(inputStr.slice(pos), 'abcdefghijk');
  });

  it('skips comments and leading newline characters when newlineOk is true', () => {
    const inputStr = '\n\n # This is a comment\n\r\n\r\nabcdefghijk';
    const pos = parseSpace(inputStr, 0, true);
    assert.equal(inputStr.slice(pos), 'abcdefghijk');
  });

  it('does not skip comments and leading newline characters when newlineOk is false', () => {
    const inputStr = '\n\n # This is a comment\n\r\n\r\nabcdefghijk';
    const pos = parseSpace(inputStr, 0, false);
    assert.equal(inputStr.slice(pos), inputStr);
  });

  it('returns an empty string if the input is all whitespace and comments', () => {
    const inputStr = '  \t# Comment\n# Another comment\n\n';
    const pos = parseSpace(inputStr, 0, true);
    assert.equal(inputStr.slice(pos), '');
  });

  it('returns an empty string for an empty input string', () => {
    const inputStr = '';
    const pos = parseSpace(inputStr, 0, true);
    assert.equal(inputStr.slice(pos), '');
  });
});

File: proposed-projects/round-trip-experiment/reverse/20260909T002206Z_273cf6b9/20260910T042804Z_e72f6c04/ported_implementation/gbnf/rules_builder/parse_space_test.py
from .parse_space import parse_space


def test_return_input_string_when_no_whitespace_or_comments():
    input = "abcdefghijk"
    assert input[parse_space(input, 0, True) :] == input


def test_skip_leading_spaces_and_tabs():
    input = "   \t   abcdefghijk"
    assert input[parse_space(input, 0, True) :] == "abcdefghijk"


def test_skip_leading_newline_characters_when_newline_ok_true():
    input = "\n\n\r\n\r\nabcdefghijk"
    assert input[parse_space(input, 0, True) :] == "abcdefghijk"


def test_not_skip_leading_newline_characters_when_newline_ok_false():
    input = "\n\n\r\n\r\nabcdefghijk"
    assert input[parse_space(input, 0, False) :] == "\n\n\r\n\r\nabcdefghijk"


def test_skip_comments_and_leading_spaces():
    input = "  # This is a comment\n\t   abcdefghijk"
    assert input[parse_space(input, 0, True) :] == "abcdefghijk"


def test_skip_comments_and_leading_newline_characters_when_newline_ok_true():
    input = "\n\n # This is a comment\n\r\n\r\nabcdefghijk"
    assert input[parse_space(input, 0, True) :] == "abcdefghijk"


def test_skip_comments_and_leading_newline_characters_when_newline_ok_false():
    input = "\n\n # This is a comment\n\r\n\r\nabcdefghijk"
    assert (
        input[parse_space(input, 0, False) :]
        == "\n\n # This is a comment\n\r\n\r\nabcdefghijk"
    )


def test_return_empty_string_if_input_all_whitespace_and_comments():
    input = "  \t# Comment\n# Another comment\n\n"
    assert input[parse_space(input, 0, True) :] == ""


def test_return_empty_string_for_empty_input_string():
    input = ""
    assert input[parse_space(input, 0, True) :] == ""
